### PR TITLE
The great API shift

### PR DESCRIPTION
**Pull request by YingboMa**
`D(f, x) * v` computes ``df/dx * v``

https://github.com/YingboMa/ForwardDiff2.jl/blob/8384e49f2fa31bf93efc837394ba59d43670d52c/README.md

### CODE REVIEW

File: src/api.jl [hunk 1]
```diff
@@ -29,35 +29,47 @@ function seed(v)
...
+    end
+    derivative = map(partials, derivative_dual)
+    is_I && (derivative *= I)
+    return derivative
+end
 
-# gradient case: f: R^n -> R
-D(sum)([1,2,3])
+function Base.:*(dd::D{<:AbstractArray}, II::UniformScaling)
+    # always chunk
```
Comment (ChrisRackauckas) [inline src/api.jl]: why?

Comment (YingboMa) [reply src/api.jl]: I didn't implement the other stuff yet. Always doing vector mode is easy to write, so I wrote that first.

Comment (ChrisRackauckas) [reply src/api.jl]: oh wait I read it wrong. Yeah, always chunk makes sense.

**Comment by YingboMa** [comment]
Waiting for https://github.com/JuliaDiff/ChainRulesCore.jl/pull/99

**Pull request merged**